src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs

```diff
@@ -41,58 +45,106 @@ private static FileStreamStrategy ChooseStrategyCore(SafeFileHandle handle, File
+                    // the code falls back to CreateFileW that just throws the right exception.
+                }
 
-            // Our Inheritable bit was stolen from Windows, but should be set in
-            // the security attributes class.  Don't leave this bit set.
-            share &= ~FileShare.Inheritable;
+                Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = GetSecAttrs(share);
 
-            // Must use a valid Win32 constant here...
-            if (mode == FileMode.Append)
-                mode = FileMode.OpenOrCreate;
+                int fAccess =
+                    ((access & FileAccess.Read) == FileAccess.Read ? Interop.Kernel32.GenericOperations.GENERIC_READ : 0) |
+                    ((access & FileAccess.Write) == FileAccess.Write ? Interop.Kernel32.GenericOperations.GENERIC_WRITE : 0);
 
-            int flagsAndAttributes = (int)options;
+                // Our Inheritable bit was stolen from Windows, but should be set in
+                // the security attributes class.  Don't leave this bit set.
+                share &= ~FileShare.Inheritable;
 
-            // For mitigating local elevation of privilege attack through named pipes
-            // make sure we always call CreateFile with SECURITY_ANONYMOUS so that the
-            // named pipe server can't impersonate a high privileged client security context
-            // (note that this is the effective default on CreateFile2)
-            flagsAndAttributes |= (Interop.Kernel32.SecurityOptions.SECURITY_SQOS_PRESENT | Interop.Kernel32.SecurityOptions.SECURITY_ANONYMOUS);
+                // Must use a valid Win32 constant here...
+                if (mode == FileMode.Append)
+                    mode = FileMode.OpenOrCreate;
 
-            using (DisableMediaInsertionPrompt.Create())
-            {
-                Debug.Assert(path != null);
-                return ValidateFileHandle(
-                    Interop.Kernel32.CreateFile(path, fAccess, share, &secAttrs, mode, flagsAndAttributes, IntPtr.Zero),
-                    path,
-                    (options & FileOptions.Asynchronous) != 0);
+                int flagsAndAttributes = (int)options;
+
+                // For mitigating local elevation of privilege attack through named pipes
+                // make sure we always call CreateFile with SECURITY_ANONYMOUS so that the
+                // named pipe server can't impersonate a high privileged client security context
+                // (note that this is the effective default on CreateFile2)
+                flagsAndAttributes |= (Interop.Kernel32.SecurityOptions.SECURITY_SQOS_PRESENT | Interop.Kernel32.SecurityOptions.SECURITY_ANONYMOUS);
+
+                SafeFileHandle safeFileHandle = ValidateFileHandle(
+                    Interop.Kernel32.CreateFile(path, fAccess, share, &secAttrs, mode, flagsAndAttributes, IntPtr.Zero), path, (options & FileOptions.Asynchronous) != 0);
```

Comment:
nit: Would you mind bringing back the original newlines? It's easier to read and doesn't make me scroll 😆:

```suggestion
                    Interop.Kernel32.CreateFile(path, fAccess, share, &secAttrs, mode, flagsAndAttributes, IntPtr.Zero),
                    path,
                    (options & FileOptions.Asynchronous) != 0);
```